cmake/ProjectMacros.cmake

```diff
@@ -62,7 +48,11 @@ macro(CREATE_TEST_TARGETS BASE_NAME SRC DEPENDENCIES USE_PCH)
 
     target_link_libraries(${BASE_NAME}_tests PRIVATE ${ALL_DEPENDENCIES} gtest)
 
-    add_google_tests(${BASE_NAME}_tests ${SRC})
+    gtest_discover_tests(${BASE_NAME}_tests
+            DISCOVERY_TIMEOUT 30
+            WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    )
+
```

Comment:
Replaced that block of "manual" source code parsing with just a call to let gtest discover the tests and add them to ctest.  Excellence.